test/unit/engine_variable.js

```diff
@@ -0,0 +1,110 @@
+const test = require('tap').test;
+const Variable = require('../../src/engine/variable');
+const htmlparser = require('htmlparser2');
+
+test('spec', t => {
+    t.type(typeof Variable.SCALAR_TYPE, typeof Variable.LIST_TYPE);
```

Comment:
I don't think this is necessary to fix now, but this test (and the one below) would probably be better written as the following:

```suggestion
    t.type(typeof Variable.SCALAR_TYPE, 'string');
```

We usually use the 'spec' test to verify the existence and types of properties on the class we are testing.

Reply:
I agree this. It must be string not only same type. 
I overlooked that the return value is used by scratch-blocks as it is.